### DISCUSSION TITLE
Perform Image Update Automation inside Helm Chart

### DISCUSSION THREAD

**Discussion by preston-willis** [fluxcd/flux2#4307] [Q&A] [answered]
I'm trying to update my image tags inside my helm repository using the image reflector, but flux isnt detecting the image policy specifier inside the chart. 

### My Question:
Are my configs wrong? 

Is this functionality supported?
If not, what is a better way to achieve this functionality?

### My ImageUpdateAutomation:
      apiVersion: image.toolkit.fluxcd.io/v1beta1
      kind: ImageUpdateAutomation
      metadata:
        name: cluster-src
        namespace: flux-system
      spec:
        git:
          commit:
            author:
              email: fluxcdbot@users.noreply.bitbucket.org
              name: fluxcdbot
            messageTemplate: "{{range .Updated.Images}}{{println .}}{{end}}"
        interval: 2m0s
        sourceRef:
          kind: GitRepository
          name: cluster-src
        update:
          path: ./app
          strategy: Setters

### My helm container yaml (./app):
      containers:
        - name: t3
          {{- if eq .Values.release "cert" }}
          image: <image> # {"$imagepolicy": "flux-system:blueline-rms-staging-policy"}
          {{- end }}
          {{- if eq .Values.release "prod" }}
          image: <image> # {"$imagepolicy": "flux-system:blueline-rms-policy"}
          {{- end }}
          ports:

**Comment by makkes** [comment]
> Is this functionality supported?

Unfortunately not because the replacement can only happen in YAML files which Helm template files are not.

> If not, what is a better way to achieve this functionality?

Most if not all charts support setting the image to use via Helm values so what you could do is use the `.spec.values` or `.spec.valuesFrom` in your HelmRelease manifest and then put the `$imagepolicy` comment marker into either the HelmRelease manifest file (for the `.spec.values` case) or into the ConfigMap/Secret (for the `.spec.valuesFrom` case).

**Reply by stefanprodan** [reply]
Here is an example of how to use image automation with HelmReleases https://fluxcd.io/flux/guides/image-update/#configure-image-update-for-custom-resources

**Reply by stefanprodan** [reply, marked as answer]
I would also like to highlight that Helm charts are meant to be pushed to a registry from where Flux downloads them. Using Flux with Helm charts from Git is highly inefficient and error prone as Flux needs to build the charts in-cluster. Please see how to push your Helm charts to a [container registry](https://helm.sh/docs/topics/registries/) like ECR and how Flux can be configured to use the charts from the registry https://fluxcd.io/flux/guides/helmreleases/#helm-oci-repository

**Reply by nwmcsween** [reply]
@stefanprodan so does Flux inherently do something different from Argocd that makes Argocd better suited for using helm charts from git?

**Reply by atharrvv** [reply]
> I would also like to highlight that Helm charts are meant to be pushed to a registry from where Flux downloads them. Using Flux with Helm charts from Git is highly inefficient and error prone as Flux needs to build the charts in-cluster. Please see how to push your Helm charts to a [container registry](https://helm.sh/docs/topics/registries/) like ECR and how Flux can be configured to use the charts from the registry https://fluxcd.io/flux/guides/helmreleases/#helm-oci-repository

IMPORTANT: But while pushing to DockerHub no matter which version i am using of Helm its returning the same
Error: push access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
despite of DockerHub Login and all!

https://docs.docker.com/docker-hub/repos/manage/hub-images/oci-artifacts/

same issue:
https://www.reddit.com/r/kubernetes/comments/1jalbkd/need_help_pushing_helm_charts_with_custom/
